### PR TITLE
Update to k8s 1.12.6 and pkg @ HEAD

### LINKED ISSUE #659: Update to support latest knative/pkg: Implentation issues such as: ClusterTask does not implement apis.Validatable
# Expected Behavior

No errors when performing a go build and using github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1 imports

# Actual Behavior

Seeing errors building ...... specifically looks like changes to knative/pkg/apis last night have caused issues for tekton

Caused by changes here :- https://github.com/knative/pkg/commit/60fdcbcabd2faeb34328d8b2725dc76c59189453

```
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/cluster_task_types.go:41:5: cannot use (*ClusterTask)(nil) (type *ClusterTask) as type apis.Validatable in assignment:
       *ClusterTask does not implement apis.Validatable (wrong type for Validate method)
               have Validate() *apis.FieldError
               want Validate(context.Context) *apis.FieldError
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/cluster_task_types.go:42:5: cannot use (*ClusterTask)(nil) (type *ClusterTask) as type apis.Defaultable in assignment:
       *ClusterTask does not implement apis.Defaultable (wrong type for SetDefaults method)
               have SetDefaults()
               want SetDefaults(context.Context)
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/pipeline_types.go:38:5: cannot use (*Pipeline)(nil) (type *Pipeline) as type apis.Validatable in assignment:
       *Pipeline does not implement apis.Validatable (wrong type for Validate method)
               have Validate() *apis.FieldError
               want Validate(context.Context) *apis.FieldError
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/pipeline_types.go:39:5: cannot use (*Pipeline)(nil) (type *Pipeline) as type apis.Defaultable in assignment:
       *Pipeline does not implement apis.Defaultable (wrong type for SetDefaults method)
               have SetDefaults()
               want SetDefaults(context.Context)
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/resource_types.go:83:5: cannot use (*PipelineResource)(nil) (type *PipelineResource) as type apis.Validatable in assignment:
       *PipelineResource does not implement apis.Validatable (wrong type for Validate method)
               have Validate() *apis.FieldError
               want Validate(context.Context) *apis.FieldError
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/resource_types.go:84:5: cannot use (*PipelineResource)(nil) (type *PipelineResource) as type apis.Defaultable in assignment:
       *PipelineResource does not implement apis.Defaultable (wrong type for SetDefaults method)
               have SetDefaults()
               want SetDefaults(context.Context)
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/task_types.go:55:5: cannot use (*Task)(nil) (type *Task) as type apis.Validatable in assignment:
       *Task does not implement apis.Validatable (wrong type for Validate method)
               have Validate() *apis.FieldError
               want Validate(context.Context) *apis.FieldError
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/task_types.go:56:5: cannot use (*Task)(nil) (type *Task) as type apis.Defaultable in assignment:
       *Task does not implement apis.Defaultable (wrong type for SetDefaults method)
               have SetDefaults()
               want SetDefaults(context.Context)
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/taskrun_types.go:30:5: cannot use (*TaskRun)(nil) (type *TaskRun) as type apis.Validatable in assignment:
       *TaskRun does not implement apis.Validatable (wrong type for Validate method)
               have Validate() *apis.FieldError
               want Validate(context.Context) *apis.FieldError
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/taskrun_types.go:31:5: cannot use (*TaskRun)(nil) (type *TaskRun) as type apis.Defaultable in assignment:
       *TaskRun does not implement apis.Defaultable (wrong type for SetDefaults method)
               have SetDefaults()
               want SetDefaults(context.Context)
vendor/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1/taskrun_types.go:31:5: too many errors
```

# Steps to Reproduce the Problem

1. Write some code that makes use of tektoncd/pipeline/pkg/apis/pipeline/v1alpha1 import
2. dep ensure to get latest levels from knative/pkg/apis
3. go build

# Additional Info

Link to slack conversation:
https://knative.slack.com/archives/CDCQ72D0R/p1553266488078200

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

I had to pin a ton of packages to get things working.  In every case I basically just pinned the version in knative/serving.

This also makes a variety of updates to the e2e tests to track changes in the upstream testing libraries.

Closes #659 

cc @mattmoor I stole your commit message :joy_cat: 
cc @ImJasonH @bobcatfish 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if functionality changed/added)
- <del>[ ] Includes [docs](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#principles) (if user facing)</del>
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
Bump kubernetes dependecies to 1.12.6
```

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -17,21 +17,21 @@ required = [
   name = "gopkg.in/yaml.v2"
   version = "v2.2.1"
 
-[[constraint]]
+[[override]]
```
Comment (vdemeester) [inline Gopkg.toml]: We need to bump `go-containerregistry` to 1.12.6 for moving those back to `constraint`

**Comment by ImJasonH** [comment]
/lgtm

Thanks! 🎉

**Comment by bobcatfish** [comment]
I think this might be a legit race condition due to changes in how logs in tests with this update:

```
I0322 17:17:40.950] ==================
I0322 17:17:40.950] WARNING: DATA RACE
I0322 17:17:40.950] Write at 0x00c0000ce138 by goroutine 68:
I0322 17:17:40.950]   flag.(*FlagSet).Parse()
I0322 17:17:40.950]       /usr/local/go/src/flag/flag.go:967 +0x42
I0322 17:17:40.951]   github.com/tektoncd/pipeline/test.initializeLogsAndMetrics()
I0322 17:17:40.951]       /usr/local/go/src/flag/flag.go:998 +0xc2
I0322 17:17:40.951]   github.com/tektoncd/pipeline/test.setup()
I0322 17:17:40.951]       /go/src/github.com/tektoncd/pipeline/test/init_test.go:48 +0xa3
I0322 17:17:40.951]   github.com/tektoncd/pipeline/test.TestGitPipelineRun()
I0322 17:17:40.951]       /go/src/github.com/tektoncd/pipeline/test/git_checkout_test.go:45 +0xdb
I0322 17:17:40.951]   testing.tRunner()
I0322 17:17:40.951]       /usr/local/go/src/testing/testing.go:865 +0x163
```

:(

**Comment by bobcatfish** [comment]
I'll do a bit of investigation, we can probably stop using those log libs in the tests entirely :innocent:

File: test/cancel_test.go [hunk 2]
```diff
@@ -136,7 +134,7 @@ func TestTaskRunPipelineRunCancel(t *testing.T) {
 		t.Errorf("Error waiting for PipelineRun `pear` to finished: %s", err)
 	}
 
-	logger.Infof("Waiting for TaskRuns in PipelineRun %s in namespace %s to be cancelled", "pear", namespace)
+	t.Logf("Waiting for TaskRuns in PipelineRun %s in namespace %s to be cancelled", "pear", namespace)
```
Comment (bobcatfish) [inline test/cancel_test.go]: there's actually one problem with using `t.Logf` in these integration tests: afaik the output is buffered, so if you run the test locally, you won't see the output until the test is done (e.g. maybe 5+ min later) :S

**Pull request merged**